crates/aptos-rosetta/src/construction.rs

```diff
@@ -627,6 +632,62 @@ async fn construction_preprocess(
     let internal_operation = InternalOperation::extract(&request.operations)?;
     let required_public_keys = vec![internal_operation.sender().into()];
 
+    if let Some(gas_price) = request.metadata.as_ref().and_then(|inner| inner.gas_price) {
+        if gas_price.0 < 1 {
+            return Err(ApiError::InvalidInput(Some(
+                "Cannot have a gas price less than 1".to_string(),
+            )));
+        }
+    }
+    if let Some(max_gas) = request
+        .metadata
+        .as_ref()
+        .and_then(|inner| inner.max_gas_amount)
+    {
+        if max_gas.0 < 1 {
```

Comment:
I wonder if we should be using non-zero types to help with this. Not worth doing now but let's open a nice-to-have issue.

Reply:
This is probably a better message than a deserialization message around the zero value.